vncviewer/Viewport.cxx

```diff
@@ -805,7 +811,9 @@ void Viewport::popupContextMenu()
   // FLTK also doesn't switch focus properly for menus
   handle(FL_UNFOCUS);
 
+  menuOpened = true;
   m = contextMenu->popup();
+  Fl::add_timeout(0.0, handleMenuClosed, this);
```

Comment:
Why the timer? Isn't the mouse press handled by the time `popup()` returns?

Reply:
No. But when the `contextMenu->popup();` returns, the mouse events that caused the popup to close are already queued (and they fire `Viewport::handle(int)`). So we queue this instant timer event after them and ignore the mouse events until the timeout event happens.
All subsequent mouse events will be queued after our timer, so they will be handled normally.